### PR TITLE
add --bam-output to gatk4_haplotypecaller

### PR DESCRIPTION
**Pull request by maxulysse**
- add --bam-output option
cf: https://gatk.broadinstitute.org/hc/en-us/articles/360037225632-HaplotypeCaller#--bam-output

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by FriederikeHanssen — APPROVED** [review]

File: modules/nf-core/gatk4/haplotypecaller/main.nf [hunk 1]
```diff
@@ -29,6 +30,7 @@ process GATK4_HAPLOTYPECALLER {
     def dbsnp_command = dbsnp ? "--dbsnp $dbsnp" : ""
     def interval_command = intervals ? "--intervals $intervals" : ""
     def dragstr_command = dragstr_model ? "--dragstr-params-path $dragstr_model" : ""
+    def bamout_command = args.contains("--bam-writer-type") ? "--bam-output ${prefix.replace('.g', '')}.realigned.bam" : ""
```
Comment (matthdsm) [inline modules/nf-core/gatk4/haplotypecaller/main.nf:33]: Isn't the `.g` suffix highly pipeline specific?

Comment (maxulysse) [reply modules/nf-core/gatk4/haplotypecaller/main.nf:33]: nope, it's pretty common actually

Comment (maxulysse) [reply modules/nf-core/gatk4/haplotypecaller/main.nf:33]: but only for vcf, this way, I'm that it's not propagated to the bam file even if specified for the vcf one

Comment (maxulysse) [reply modules/nf-core/gatk4/haplotypecaller/main.nf:33]: To be fair, the perfect solution would be to have 2 prefixes, but that sounds like crazy

Comment (matthdsm) [reply modules/nf-core/gatk4/haplotypecaller/main.nf:33]: True that!

Comment (matthdsm) [reply modules/nf-core/gatk4/haplotypecaller/main.nf:33]: or if `prefix` is something file `name.genomex`

Comment (maxulysse) [reply modules/nf-core/gatk4/haplotypecaller/main.nf:33]: it does just replace `.g` by nothing, could be more efficient by removing `.g` if it's at the end of the string indeed, that would solve this `sample.gnarly_name` issue

Comment (maxulysse) [reply modules/nf-core/gatk4/haplotypecaller/main.nf]: now fixed

**Pull request merged**